### PR TITLE
`split -1 in` should be accepted

### LINKED ISSUE #5033: `split -1 in` should be accepted
```
touch in
./target/debug/coreutils split -1 in
```
=> 
```
error: unexpected argument '-1' found

  tip: to pass '-1' as a value, use '-- -1'
```

It should be equivalent to `-a 1`

### PR DESCRIPTION
**Pull request by zhitkoff**
It looks like the GNU `split -1 in` behaves as `split -l 1 in` , i.e. a shorthand for `-l` or `--lines`
Having that, here is my take on it - using similar approach to what is done in `fold`

fixes #5033

### CODE REVIEW

File: src/uu/split/src/split.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
-//  * This file is part of the uutils coreutils package.
-//  *
 //  * (c) Akira Hayakawa <ruby.wktk@gmail.com>
 //  *
 //  * For the full copyright and license information, please view the LICENSE
 //  * file that was distributed with this source code.
+//  *
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:5]: please revert this

**Comment by sylvestre** [comment]
a few comments:
* could you please run rustfmt?
* it seems that it regressed some tests
* we will need a test covering this change
thanks

**Comment by tertsdiepraam** [comment]
In the issue I wrote that it should behave like `fold` (though we can use another method). Have you compared this method with what we do in `fold`?

File: src/uu/split/src/split.rs (hunk 1 above)
Comment (zhitkoff) [reply src/uu/split/src/split.rs:5]: yep, will do

**Comment by zhitkoff** [comment]
> In the issue I wrote that it should behave like `fold` (though we can use another method). Have you compared this method with what we do in `fold`?

I see ... So, basically - instead of trying to handle it all with clap::Arg , strip out the "obsolete" arguments like this one and handle it separately. Will do that. 
However, one note on that - the implementation in `fold` allows both "obsolete" and "normal" arguments for width to be specified at the same time, but if so, only "normal" one will be used. 
That is correct behavior for GNU `fold` , but it is not for GNU `split` - it fails when both used like so `split -9 --lines=9 file`. So, implementation would need to be a bit different for `split`

**Comment by tertsdiepraam** [comment]
> So, basically - instead of trying to handle it all with clap::Arg , strip out the "obsolete" arguments like this one and handle it separately. Will do that.

Yes, that's the best approach we found so far, but if you find a better way do let us know!

> That is correct behavior for GNU fold , but it is not for GNU split - it fails when both used like so split -9 --lines=9 file. So, implementation would need to be a bit different for split

`split` technically works like `uniq`, not like `fold`, but we simplified it a bit in `uniq`. It's very tricky to support all of these properly. I think that, for now, we have to find a balance between being compatible and having reasonably simple code. It just does not fit `clap`s API very well.

**Comment by zhitkoff** [comment]
> GNU testsuite comparison:
> 
> ```
> Skip an intermittent issue tests/tail-2/inotify-dir-recreate
> ```

This one actually shows as FAIL in gnu-full-result.json artifact from the last GnuTests action run for this PR.
Not sure what can be done to fix this ... considering this PR did not touch `tail` or any common code.
@tertsdiepraam @sylvestre could you please point me in the right direction?

**Comment by zhitkoff** [comment]
> a few comments:
> 
> * could you please run rustfmt?
> * it seems that it regressed some tests
> * we will need a test covering this change
>   thanks

should be all done, except some weird issue with GnuTests regression on tests/tail-2/inotify-dir-recreate ... see my other comment https://github.com/uutils/coreutils/pull/5170#issuecomment-1688490222

**Comment by zhitkoff** [comment]
rebased to main

**Comment by zhitkoff** [comment]
as noted in issue #5033:
This is a bit more complicated than supporting split -1 in use case. The -1 obsolete option could also be found inside combined short options, like following (all below are valid and work with GNU split)
split -1dxe file which means split -l 1 -d -x -e file
split -ex23d file which means split -e -x -l 23 -d file
etc
will need to refactor, so I will close this pull request for now and resubmit a new one with code that would handle this broader use case.

**Pull request closed without merging**